web/src/features/moreCast2/components/TabbedDataGrid.tsx

```diff
@@ -496,10 +500,12 @@ const TabbedDataGrid = ({ fromTo, setFromTo, fetchWeatherIndeterminates }: Tabbe
 
   // Handler passed to our datagrids that runs after a row is updated.
   const processRowUpdate = (newRow: MoreCast2Row) => {
-    const filledRows = fillStationGrassCuringForward(newRow, allRows)
+    let filledRows = fillStationGrassCuringForward(newRow, allRows)
     const filteredRows = filterRowsForSimulationFromEdited(newRow, filledRows)
     const filteredRowsWithIndices = simulateFireWeatherIndices(filteredRows)
-    storedDraftForecast.updateStoredDraftForecasts(filteredRowsWithIndices, getDateTimeNowPST())
+    // Merge rows that have been updated (ie. filledRows) with rows that have had indices added and save as a draft
+    filledRows = filledRows.map(filledRow => filteredRowsWithIndices.find(row => row.id === filledRow.id) || filledRow)
```

Comment:
Could filledRows and newRows be the same? Can we get rid of one of them?

Reply:
They need to be separate as the `filledRows` are the visible rows that have been edited and need to be saved to local storage. `newRows` represents all the rows in the datagrid, not just the edited ones. I actually need to update line 511 to update `newRows` from `filledRows` instead of from `filteredRowsWithIndices`.